controllers/cluster_create.go

```diff
@@ -1213,9 +1219,9 @@ func electPvcToReattach(cluster *apiv1.Cluster) string {
 		return ""
 	}
 
-	for _, name := range pvcs {
-		if name == cluster.Status.TargetPrimary {
-			return name
+	for _, pvc := range pvcs {
+		if specs.DoesBelongToInstance(cluster.Status.TargetPrimary, pvc) {
+			return pvc
```

Comment:
I think the function `electPvcToReattach` is not suitable anymore, as we may want to reattach two PVCs: the pgdata one and the WALs one. Maybe we could change its name as `electInstanceToReattach` and make sure we always return the name of the instance, which corresponds to the name of the pgdata Pod.